pkg/util/deployment/deployment.go

```diff
@@ -30,22 +30,34 @@ import (
 
 // Returns the old RCs targetted by the given Deployment.
 func GetOldRCs(deployment extensions.Deployment, c client.Interface) ([]*api.ReplicationController, error) {
+	return GetOldRCsCustomized(deployment, c,
+		func(namespace string) (*api.PodList, error) {
+			return c.Pods(namespace).List(labels.SelectorFromSet(deployment.Spec.Selector), fields.Everything())
+		},
+		func(namespace string) ([]api.ReplicationController, error) {
+			rcList, err := c.ReplicationControllers(namespace).List(labels.Everything(), fields.Everything())
+			return rcList.Items, err
+		})
+}
+
+// Returns the old RCs targetted by the given Deployment using customized functions to get PodList and RCList.
+func GetOldRCsCustomized(deployment extensions.Deployment, c client.Interface, getPodList func(string) (*api.PodList, error), getRcList func(string) ([]api.ReplicationController, error)) ([]*api.ReplicationController, error) {
```

Comment:
@nikhiljindal Since there are two ways to `GetOldRCs`, I use this function to separate both ways (via client vs. via dc.podStore/rcStore). So as `GetNewRC`.